controllers/openstackmachine_controller.go

```diff
@@ -147,12 +147,18 @@ func (r *OpenStackMachineReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return reconcile.Result{}, err
 	}
 
-	// Resolve and store referenced resources
+	// Resolve and store referenced OpenStack resources
 	err = compute.ResolveReferencedMachineResources(scope, &openStackMachine.Spec, &openStackMachine.Status.ReferencedResources)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
+	// Resolve referenced resources CAPO resources, using the K8s client
+	err = resolveReferencedClientResources(ctx, r.Client, openStackMachine)
```

Comment:
It feels like it's still a Machine resource. Couldn't we put that into `ResolveReferencedMachineResources` directly? Even if we need to change the arguments of the function.

Reply:
I see that, sure. Let's see what other reviewers will say here, especially @mdbooth as `ResolveReferencedMachineResources()` is an idea of his.

Reply:
While rebased onto `v1beta` I have removed `resolveReferencedClientResources` and moved the code into the new `resolveMachineResources`.